File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        st=s.strip()
        i=len(st)-1
        mid=""
        res=""
        while i>=0:
            
            if st[i]==' ':
                if mid!="" and mid!=" ":
                    res+=(mid[::-1]+' ')
                    mid=""
            else:
                mid+=st[i]
            i-=1
        res+=(mid.strip()[::-1])
        return res.strip()



        

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        int i=0,j=s.size()-1;
        while(i<=j && s[i]==' ') i++;
        while(j>=0 && s[j]==' ') j--;
        string ans="";
        int k=j;
        for(;k>=i;){
            while(k>=0 && s[k]==' ') k--;
            string t="";
            while(k>=0 && s[k]!=' '){
                t=s[k]+t;
                k--;
            }
            if(!ans.empty()) ans+=" ";
            ans+=t;
        }
        return ans;
    }
};